### PR TITLE
Event Join

### PR DESCRIPTION
**Pull request by zyuiop**
- [x] Join an event clicking a toggle
- [x] Save stored events in a SQLite database
- [x] Display joined events before the rest in the event selection screen (issue #36)

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/JoinedEvent.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        JoinedEvent that = (JoinedEvent) o;
+
+        if (uid != that.uid) return false;
+        return name != null ? name.equals(that.name) : that.name == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = uid;
+        result = 31 * result + (name != null ? name.hashCode() : 0);
```
Comment (Fnux) [inline app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/JoinedEvent.java:83]: Where does this magic `31` comes from?

File: app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventSelector/EventPickingModel.java [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-import android.arch.lifecycle.ViewModelProvider;
 import ch.epfl.sweng.eventmanager.repository.EventRepository;
 import ch.epfl.sweng.eventmanager.repository.data.Event;
-import dagger.Binds;
+import ch.epfl.sweng.eventmanager.repository.JoinedEventRepository;
 
 import javax.inject.Inject;
 
-import java.util.List;
+import java.util.*;
```
Comment (Fnux) [inline app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventSelector/EventPickingModel.java]: That's a lot of things.

File: app/src/main/res/layout/activity_event.xml [hunk 3]
```diff
@@ -98,4 +98,12 @@
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/event_image"
         app:layout_constraintVertical_bias="0.977" />
+    <Switch
+            android:text="Join the event"
```
Comment (Fnux) [inline app/src/main/res/layout/activity_event.xml]: We should avoid hardcoded strings as much as possible.

File: app/src/main/res/layout/activity_event_picking.xml [hunk 4]
```diff
@@ -1,33 +1,56 @@
 <?xml version="1.0" encoding="utf-8"?>
 <android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".ui.eventSelector.EventPickingActivity">
+                                             xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (Fnux) [inline app/src/main/res/layout/activity_event_picking.xml]: That's not how the other XML files were indented. I don't mind this one, but we have to stay coherent and use the same indentation scheme everywhere!

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1,4 +1,7 @@
 <resources>
     <string name="app_name">Event Manager</string>
     <string name="title_activity_event_picking">EventPickingActivity</string>
+    <string name="picking_activity_help_text">Please select an event to continue.</string>
```
Comment (Fnux) [inline app/src/main/res/values/strings.xml:3]: Please be careful on naming convention:
  * `title_activity_event_picking`
  * `picking_activity_help_text`

... use somewhat reversed naming schemes. We have to choose one and stick to it.

**Review by Fnux — COMMENTED** [review]
Good job!

File: app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventSelector/EventPickingModel.java (hunk 2 above)
Comment (antoine-crettenand) [reply app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventSelector/EventPickingModel.java]: Intellij auto-import feature does this I guess. It's fixed.

File: app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/JoinedEvent.java (hunk 1 above)
Comment (antoine-crettenand) [reply app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/JoinedEvent.java:83]: I found it here : [https://en.wikipedia.org/wiki/Java_hashCode()#The_java.lang.String_hash_function](https://en.wikipedia.org/wiki/Java_hashCode()#The_java.lang.String_hash_function)

File: app/src/main/res/layout/activity_event.xml (hunk 3 above)
Comment (antoine-crettenand) [reply app/src/main/res/layout/activity_event.xml]: Fixed.

File: app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/JoinedEvent.java (hunk 1 above)
Comment (Fnux) [reply app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/JoinedEvent.java:83]: Please briefly explain from where it comes from in a comment.

**Review by Fnux — APPROVED** [review]

**Pull request merged**